WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -2484,12 +2485,32 @@ protected void setPostContentFromShareAction() {
         }
     }
 
-    private void setFeaturedImageId(final long mediaId, final boolean imagePicked) {
-        if (mEditPostSettingsFragment != null) {
-            mEditPostSettingsFragment.updateFeaturedImage(mediaId, imagePicked);
-            if (mEditorFragment instanceof GutenbergEditorFragment) {
-                ((GutenbergEditorFragment) mEditorFragment).sendToJSFeaturedImageId((int) mediaId);
+    private void setFeaturedImageId(final long mediaId, final boolean imagePicked, final boolean isGutenbergEditor) {
+        if (isGutenbergEditor) {
+            EditPostRepository postRepository = getEditPostRepository();
+            if (postRepository == null) {
+                return;
             }
+
+            int postId = getEditPostRepository().getId();
+            if (mediaId == 0) {
```

Comment:
@SiobhyB wdyt about importing `MEDIA_ID_NO_FEATURED_IMAGE_SET ` here?

Reply:
@jd-alexander, that makes sense to me! I've gone ahead to import that constant in https://github.com/wordpress-mobile/WordPress-Android/pull/14503/commits/fb3ba9548da20bea53f50ca56e8bc6fc584e877e. Let me know if anything looks off to you. :)

Reply:
Looks good to me!